thrall/app/lib/RetryHandler.scala

```diff
@@ -43,12 +44,13 @@ object RetryHandler {
       Future.firstCompletedOf(Seq(timeout, f()))
     }
 
-    def handleWithRetry[T](f: () => Future[T], retries: Int, delay: FiniteDuration): () => Future[T] = () => {
+    def handleWithRetry[T](f: () => Future[T], retries: Int, delay: FiniteDuration)(implicit markerContext: MarkerContext): () => Future[T] = () => {
       implicit val scheduler = actorSystem.scheduler
       var count = 0
 
       def attempt = () => {
         count = count + 1
+        MarkerAugmentation.augmentMarkerContext(Map("attempt" -> count))
```

Comment:
Could we be more explicit here?

```suggestion
        MarkerAugmentation.augmentMarkerContext(Map("retryCount" -> count))
```